### PR TITLE
RewardsDistributor management improvements + new features

### PR DESCRIPTION
**Pull request by dbeal-eth**
need to write tests and etc but this look like it will work

### CODE REVIEW

File: protocol/synthetix/contracts/modules/core/RewardsManagerModule.sol
Comment (noisekit) [inline protocol/synthetix/contracts/modules/core/RewardsManagerModule.sol:87]: do we expect `else` here?

File: protocol/synthetix/contracts/storage/Pool.sol
Comment (noisekit) [inline protocol/synthetix/contracts/storage/Pool.sol:403]: As `updateRewardsToVaults` gets called for any vault "action" this is going to increase gas by a bit for every vault added to the pool? So the more collaterals we have the more and more gas we are going to consume for every delegation change

File: protocol/synthetix/contracts/modules/core/RewardsManagerModule.sol
Comment (dbeal-eth) [reply protocol/synthetix/contracts/modules/core/RewardsManagerModule.sol:87]: Yes

File: protocol/synthetix/contracts/storage/Pool.sol
Comment (dbeal-eth) [reply protocol/synthetix/contracts/storage/Pool.sol:403]: No the loop is on number of RDs in the pool, not number if collateral types/vaults in the pool. so the gas increase is caused by the addition of any pool level RDs, which is inevitable considering this is the ask for the change

Comment (noisekit) [reply protocol/synthetix/contracts/storage/Pool.sol:403]: Yeah sure. I meant to write rd not collateral.

Comment (noisekit) [reply protocol/synthetix/contracts/storage/Pool.sol:403]: would be very good to see a gas impact for arbitrum and base deployments (with 4 and 2 distributors deployed on those).

**Review by noisekit — APPROVED** [review]
This looks good to me. Approved, but please get the auditors approval before merge.

File: protocol/synthetix/test/integration/modules/core/RewardsManagerModule.test.ts
Comment (noisekit) [inline protocol/synthetix/test/integration/modules/core/RewardsManagerModule.test.ts:705]: Asert?

File: auxiliary/RewardsDistributor/cannonfile.toml
Comment (noisekit) [inline auxiliary/RewardsDistributor/cannonfile.toml:41]: Can we keep args without breaking change and accept zero address instead please

File: auxiliary/RewardsDistributor/src/RewardsDistributor.sol
Comment (noisekit) [inline auxiliary/RewardsDistributor/src/RewardsDistributor.sol:22]: We still want to have per collateral distributors available. Please dont remove this functionality

File: protocol/synthetix/test/integration/modules/core/RewardsManagerModule.test.ts
Comment (dbeal-eth) [reply protocol/synthetix/test/integration/modules/core/RewardsManagerModule.test.ts]: sorry no that was just for sanity debug. I removed the console log and the getavailablerewards call

File: auxiliary/RewardsDistributor/cannonfile.toml
Comment (dbeal-eth) [reply auxiliary/RewardsDistributor/cannonfile.toml:41]: its not a breaking change because its for the constructor of the contract, and the functionality of the contract has significantly changed in an of itself. It might as well be a new contract. since the contract no longer pays any attention to collateral type, I went ahead and removed the arg. On the contrary, I think this should be useful in creating awareness of the change downstream in case its forgotten about.

File: auxiliary/RewardsDistributor/src/RewardsDistributor.sol
Comment (dbeal-eth) [reply auxiliary/RewardsDistributor/src/RewardsDistributor.sol:22]: before removal, the storage of the value here was only for verification against when calling the `distributeRewards` function. the `distributeRewards` function below still takes in `collateralType` as an argument, allowing for collateral to be distributed to any vault in the pool or at the pool which it is registered to on the core system. since it was only used for that verification before, the storage has been removed because the value isnt used anywhere anymore.

Comment (noisekit) [reply auxiliary/RewardsDistributor/src/RewardsDistributor.sol:22]: Ok makes sense thanks 🙏

File: auxiliary/RewardsDistributor/cannonfile.toml
Comment (noisekit) [reply auxiliary/RewardsDistributor/cannonfile.toml:41]: Got it, thanks

**Review by noisekit — APPROVED** [review]

File: protocol/synthetix/contracts/modules/core/RewardsManagerModule.sol
Comment (noisekit) [inline protocol/synthetix/contracts/modules/core/RewardsManagerModule.sol:319]:
```suggestion
```

File: protocol/synthetix/contracts/storage/Pool.sol
Comment (noisekit) [inline protocol/synthetix/contracts/storage/Pool.sol:402]:
```suggestion
```

**Comment by dbeal-eth** [comment]
auditor has greenlit

**Pull request merged**
